loki/ir/nodes.py

```diff
@@ -501,7 +501,14 @@ def association_map(self):
         """
         An :any:`collections.OrderedDict` of associated expressions.
         """
-        return OrderedDict(self.associations)
+        return CaseInsensitiveDict((str(k), v) for k, v in self.associations)
+
+    @property
+    def inverse_map(self):
+        """
+        An :any:`collections.OrderedDict` of associated expressions.
+        """
+        return CaseInsensitiveDict((str(v), k) for k, v in self.associations)
```

Comment:
Can we simply "stringify" `v` here, considering that an association can in theory be a rather complex expression?